lib/fastlane/plugin/wpmreleasetoolkit/helper/glotpress_download_helper.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'uri'
+
+module Fastlane
+  module Helper
+    # A helper class to download files from GlotPress with proper error handling and retry mechanism
+    class GlotPressDownloader
+      AUTO_RETRY_SLEEP_TIME = 20
+      MAX_AUTO_RETRY_ATTEMPTS = 30
+
+      attr_reader :auto_retry, :auto_retry_attempt_counter
+
+      # Initialize a new GlotPressDownloader
+      #
+      # @param [Boolean] auto_retry Whether to automatically retry on rate limiting (429 errors)
+      #
+      def initialize(auto_retry: false)
+        @auto_retry = auto_retry
+        @auto_retry_attempt_counter = 0
+      end
+
+      # Downloads data from GlotPress
+      #
+      # @param [String] url The URL to download from
+      # @param [String] locale The locale being downloaded (for logging purposes)
+      # @yield [String] The response body if the download was successful
+      #
+      def download(url, locale)
+        @current_locale = locale # Store for error handling
+        uri = URI(url)
+        response = make_request(uri)
+        handle_response(response: response, locale: locale, original_uri: uri) do |body|
+          yield body if block_given?
+        end
+      end
+
+      # Downloads data from GlotPress and returns the response body
+      #
+      # @param [String] url The URL to download from
+      # @param [String] locale The locale being downloaded (for logging purposes)
+      # @return [String, nil] The response body if successful, nil otherwise
+      #
+      def download_and_return(url, locale)
+        result = nil
+        download(url, locale) { |body| result = body }
+        result
+      end
+
+      private
+
+      def make_request(uri)
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = (uri.scheme == 'https')
+        request = Net::HTTP::Get.new(uri.request_uri)
+        request['User-Agent'] = Wpmreleasetoolkit::USER_AGENT
+        http.request(request)
+      rescue StandardError => e
+        # Network errors, connection errors, etc.
+        UI.error("Error downloading locale `#{@current_locale}` — #{e.message} (#{uri})")
+        retry if !FastlaneCore::Helper.is_ci? && UI.confirm("Retry downloading `#{@current_locale}`?")
```

Comment:
This means that if we're on CI it won't retry though, right?

While I think we want to auto-retry if CI is `true`, _or_ if CI is false but we `UI.confirm`'d we want to retry?
So shouldn't this be:

```suggestion
        retry if FastlaneCore::Helper.is_ci? || UI.confirm("Retry downloading `#{@current_locale}`?")
```

Reply:
Or is this only for retries that are unrelated to 429 and other cases that you handle in `handle_response`?

Reply:
Yeah I think this is a case where we want CI to fail, unrelated to 429.